# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../gnupg/sparc64.patch
# Copyright (C) 2004 - 2006 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Instrumented the sparc assembly to compile for sparc64 with recent binutils
versions ... The check does bus-error, though. But it does with generic C
code, too. So it is more likely some other problem ...

(Cross review libgcrypt/sparc64.patch while performing changes.)

  - Rene Rebe <rene@exactcode.de>

diff -ur gnupg-1.2.6/mpi/sparc32/mpih-add1.S gnupg-1.2.6-fixed/mpi/sparc32/mpih-add1.S
--- gnupg-1.2.6/mpi/sparc32/mpih-add1.S	2002-06-29 16:14:10.000000000 +0200
+++ gnupg-1.2.6-fixed/mpi/sparc32/mpih-add1.S	2005-01-23 07:23:35.000000000 +0100
@@ -39,6 +39,9 @@
 
 #include "sysdep.h"
 
+	.register %g2,#scratch
+	.register %g3,#scratch
+
 	.text
 	.align	4
 	.global C_SYMBOL_NAME(mpihelp_add_n)
diff -ur gnupg-1.2.6/mpi/sparc32/mpih-lshift.S gnupg-1.2.6-fixed/mpi/sparc32/mpih-lshift.S
--- gnupg-1.2.6/mpi/sparc32/mpih-lshift.S	2002-06-29 16:14:10.000000000 +0200
+++ gnupg-1.2.6-fixed/mpi/sparc32/mpih-lshift.S	2005-01-23 07:23:46.000000000 +0100
@@ -29,6 +29,9 @@
 
 #include "sysdep.h"
 
+	.register %g2,#scratch
+	.register %g3,#scratch
+
 	.text
 	.align	4
 	.global C_SYMBOL_NAME(mpihelp_lshift)
diff -ur gnupg-1.2.6/mpi/sparc32/mpih-rshift.S gnupg-1.2.6-fixed/mpi/sparc32/mpih-rshift.S
--- gnupg-1.2.6/mpi/sparc32/mpih-rshift.S	2002-06-29 16:14:10.000000000 +0200
+++ gnupg-1.2.6-fixed/mpi/sparc32/mpih-rshift.S	2005-01-23 07:23:59.000000000 +0100
@@ -28,6 +28,9 @@
 
 #include "sysdep.h"
 
+	.register %g2,#scratch
+	.register %g3,#scratch
+
 	.text
 	.align	4
 	.global C_SYMBOL_NAME(mpihelp_rshift)
diff -ur gnupg-1.2.6/mpi/sparc32/udiv.S gnupg-1.2.6-fixed/mpi/sparc32/udiv.S
--- gnupg-1.2.6/mpi/sparc32/udiv.S	2002-06-29 16:14:10.000000000 +0200
+++ gnupg-1.2.6-fixed/mpi/sparc32/udiv.S	2005-01-23 07:24:16.000000000 +0100
@@ -38,6 +38,9 @@
 
 #include "sysdep.h"
 
+	.register %g2,#scratch
+	.register %g3,#scratch
+
 	.text
 	.align 4
 	.global C_SYMBOL_NAME(__udiv_qrnnd)
diff -ur gnupg-1.2.6/mpi/sparc32v8/mpih-mul1.S gnupg-1.2.6-fixed/mpi/sparc32v8/mpih-mul1.S
--- gnupg-1.2.6/mpi/sparc32v8/mpih-mul1.S	2002-06-29 16:14:10.000000000 +0200
+++ gnupg-1.2.6-fixed/mpi/sparc32v8/mpih-mul1.S	2005-01-23 07:22:22.000000000 +0100
@@ -39,6 +39,9 @@
 
 #include "sysdep.h"
 
+	.register %g2,#scratch
+	.register %g3,#scratch
+
 .text
 	.align	8
 	.global C_SYMBOL_NAME(mpihelp_mul_1)
diff -ur gnupg-1.2.6/mpi/sparc32v8/mpih-mul2.S gnupg-1.2.6-fixed/mpi/sparc32v8/mpih-mul2.S
--- gnupg-1.2.6/mpi/sparc32v8/mpih-mul2.S	2002-06-29 16:14:10.000000000 +0200
+++ gnupg-1.2.6-fixed/mpi/sparc32v8/mpih-mul2.S	2005-01-23 07:22:46.000000000 +0100
@@ -39,6 +39,9 @@
 
 #include "sysdep.h"
 
+	.register %g2,#scratch
+	.register %g3,#scratch
+
 .text
 	.align 4
 	.global C_SYMBOL_NAME(mpihelp_addmul_1)
diff -ur gnupg-1.2.6/mpi/sparc32v8/mpih-mul3.S gnupg-1.2.6-fixed/mpi/sparc32v8/mpih-mul3.S
--- gnupg-1.2.6/mpi/sparc32v8/mpih-mul3.S	2002-06-29 16:14:10.000000000 +0200
+++ gnupg-1.2.6-fixed/mpi/sparc32v8/mpih-mul3.S	2005-01-23 07:22:59.000000000 +0100
@@ -38,6 +38,9 @@
 
 #include "sysdep.h"
 
+	.register %g2,#scratch
+	.register %g3,#scratch
+
 .text
 	.align 4
 	.global C_SYMBOL_NAME(mpihelp_submul_1)
